apps/arena/lib/arena/game_updater.ex

```diff
@@ -532,27 +532,27 @@ defmodule Arena.GameUpdater do
         %{game_state: game_state, game_config: game_config} = state
       ) do
     entry = %{killer_id: killer_id, victim_id: victim_id}
-    victim = Map.get(game_state.players, victim_id)
-    killer = Map.get(game_state.players, killer_id)
 
     game_state =
       game_state
       |> update_in([:killfeed], fn killfeed -> [entry | killfeed] end)
       |> maybe_add_kill_to_player(killer_id)
-      |> grant_power_up_to_killer(game_config, killer, victim)
```

Comment:
The problem here is that `maybe_add_kill_to_player/2` was updating the `killer` inside `game_state`, but then `grant_power_up_to_killer/3` was using the old `killer` binded outside the pipeping, in essence, overriding any changes that happened to `killer` after that binding

Reply:
Nice catch!

Reply:
Immutable data played all of us while reviewing 😆